libyul/optimiser/CommonSubexpressionEliminator.cpp

```diff
@@ -103,25 +105,44 @@ void CommonSubexpressionEliminator::visit(Expression& _e)
 					_e = Identifier{debugDataOf(_e), value->name};
 		}
 	}
-	else
-	{
-		// TODO this search is rather inefficient.
-		for (auto const& [variable, value]: allValues())
-		{
-			assertThrow(value.value, OptimizerException, "");
-			// Prevent using the default value of return variables
-			// instead of literal zeros.
-			if (
-				m_returnVariables.count(variable) &&
-				holds_alternative<Literal>(*value.value) &&
-				valueOfLiteral(get<Literal>(*value.value)) == 0
-			)
-				continue;
-			if (SyntacticallyEqual{}(_e, *value.value) && inScope(variable))
+	else if (
+		auto it = m_replacementCandidates.find(&_e);
+		it != m_replacementCandidates.end()
+	)
```

Comment:
Could probably use something like:
```suggestion
	else if (auto const* replacementCandidate = util::valueOrNullptr(m_replacementCandidates, &_e))
```